app/code/core/Mage/Core/Helper/Purifier.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * OpenMage
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available at https://opensource.org/license/osl-3-0-php
+ *
+ * @category   Mage
+ * @package    Mage_Core
+ * @copyright  Copyright (c) 2023 The OpenMage Contributors (https://www.openmage.org)
+ * @license    https://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+/**
+ * HTMLPurifier helper
+ *
+ * @category   Mage
+ * @package    Mage_Core
+ */
+class Mage_Core_Helper_Purifier extends Mage_Core_Helper_Abstract
+{
+    public const CACHE_DEFINITION = 'Cache.DefinitionImpl';
+
+    /**
+     * @var HTMLPurifier|null
+     */
+    protected ?HTMLPurifier $purifier;
+
+    /**
+     * Purifier Constructor Call
+     * @param HTMLPurifier|null $purifier
+     */
+    public function __construct(
+        ?HTMLPurifier $purifier = null
+    ) {
+        $config = HTMLPurifier_Config::createDefault();
+        $config->set(self::CACHE_DEFINITION, null);
+        $this->purifier = $purifier ?? new HTMLPurifier($config);
+    }
+
+    /**
+     * Purify Html Content
+     *
+     * @param array|string $content
+     * @return array|string
+     */
+    public function purify(array|string $content): array|string
```

Comment:
ahhh the PHP8 problem is just about this line, not an incompatibility with htmlpurify itself, so it should be easier to solve